// Copyright 2024 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package mediapipe;

message ResourceProviderCalculatorOptions {
  // The calculator will use `resource_id` to load and provide the resource
  // through `CalculatorContext::GetResources`.
  //
  // Resource ID can be a file path, but is not limited to just that. (Clients
  // can override resource loading through `kResourcesService`)
  repeated string resource_id = 1;

  enum ReadMode {
    READ_MODE_UNDEFINED = 0;  // Defaults to BINARY
    READ_MODE_BINARY = 1;
    READ_MODE_TEXT = 2;
  }

  ReadMode read_mode = 2;
}
